### PR TITLE
Modal for missing vaccine data

### PR DESCRIPTION
**Pull request by bionara**
# Description

Adds a popup message for when the user must edit their vaccine data. Edit button links to vaccine edit page.

To test: Add a vaccine, then delete its name/description in the django admin. Return to page, refresh, tap bottom CTA to trigger modal.

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

image

## Checklist

- [ ] I have updated mockServer
- [ ] I've added analytics as relevent
- [ ] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up

_Is there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_

### CODE REVIEW

File: src/CovidApp.tsx [hunk 1]
```diff
@@ -51,6 +52,11 @@ function CovidApp() {
             component={ShareScreen}
             options={{ cardStyle: { backgroundColor: 'rgba(0,0,0,0.8)' } }}
           />
+          <Stack.Screen
+            name="VaccineListMissing"
+            component={VaccineListMissingModal}
+            options={{ cardStyle: { backgroundColor: 'rgba(0,0,0,0.4)' } }}
```
Comment (petermason) [inline src/CovidApp.tsx:58]: shall we make the backgroundColor the same for each?

File: src/routes/main-navigator/index.tsx [hunk 2]
```diff
@@ -79,13 +80,12 @@ import {
...
 } from '@covid/features';
+import { VaccineListMissingModal } from '@covid/features/vaccines/VaccineListMissingModal';
 
 import DietStudyPlaybackNavigator from '../diet-study-playback-navigator';
-import MentalHealthNavigator from '../main-navigator';
-import { MetalHealthNavigator } from '..';
 
 function MainNavigator() {
-  const Stack = createStackNavigator<ScreenParamList>();
+  const Stack: TypedNavigator<NavigationStackProp> = createStackNavigator<ScreenParamList>();
```
Comment (petermason) [inline src/routes/main-navigator/index.tsx:88]: just curious, what's the additional typing for?

File: src/features/vaccines/VaccineListMissingModal.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { RouteProp } from '@react-navigation/native';
+import { StackNavigationProp } from '@react-navigation/stack';
+import { Text } from 'native-base';
```
Comment (petermason) [inline src/features/vaccines/VaccineListMissingModal.tsx:3]: better to use Text from '@covid/components'

File: src/features/vaccines/VaccineListMissingModal.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import React from 'react';
+import { StyleSheet, View } from 'react-native';
+
+import { colors } from '@theme';
+import i18n from '@covid/locale/i18n';
+import { ScreenParamList } from '@covid/features/ScreenParamList';
+import { HeaderText, BrandedButton } from '@covid/components';
```
Comment (petermason) [inline src/features/vaccines/VaccineListMissingModal.tsx:10]: better to use Text from '@covid/components'

File: src/CovidApp.tsx (hunk 1 above)
Comment (bionara) [reply src/CovidApp.tsx:58]: yep, can i update the share one to be more transparent? maybe 0.5 for both

File: src/features/vaccines/VaccineListMissingModal.tsx (hunk 3 above)
Comment (bionara) [reply src/features/vaccines/VaccineListMissingModal.tsx:3]: good catch, updated!

File: src/routes/main-navigator/index.tsx (hunk 2 above)
Comment (bionara) [reply src/routes/main-navigator/index.tsx:88]: i'm actually not sure how that got in! maybe vs code auto complete? removing

File: src/routes/main-navigator/index.tsx [hunk 5]
```diff
@@ -180,7 +180,6 @@ function MainNavigator() {
       <Stack.Screen name="SearchLAD" component={SearchLADScreen} options={noHeader} />
       {/* __HYGEN_INJECTED_SCREEN_ABOVE__ */}
       {DietStudyPlaybackNavigator({ Stack })}
-      {MetalHealthNavigator({ Stack })}
```
Comment (petermason) [inline src/routes/main-navigator/index.tsx:183]: has this been deleted?

Comment (bionara) [reply src/routes/main-navigator/index.tsx:183]: sorry, looked unused and suffered th wrath of my house keeping. will be more attentive in future. restored.

File: src/CovidApp.tsx (hunk 1 above)
Comment (petermason) [reply src/CovidApp.tsx]: What's in Figma?

**Review by petermason — APPROVED** [review]

**Pull request merged**